tests/integration_tests/modules/test_lxd.py

```diff
@@ -62,12 +137,52 @@ def validate_storage(validate_client, pkg_name, command):
     return log
 
 
+def validate_preseed_profiles(client, preseed_cfg):
+    for src_profile in preseed_cfg["profiles"]:
+        profile = yaml.safe_load(
+            client.execute(f"lxc profile show {src_profile['name']}")
+        )
+        assert src_profile["config"] == profile["config"]
+
+
+def validate_preseed_storage_pools(client, preseed_cfg):
+    for src_storage in preseed_cfg["storage_pools"]:
+        storage_pool = yaml.safe_load(
+            client.execute(f"lxc storage show {src_storage['name']}")
+        )
+        assert storage_pool["config"]["source"] == storage_pool["config"].pop(
+            "volatile.initial_source"
+        )
+        assert storage_pool["config"] == src_storage["config"]
```

Comment:
While testing in `kinetic`, the config for the `test_storage_preseed_zfs` case has an extra key:

```yaml
config:
  size: 5GiB
  source: /var/snap/lxd/common/lxd/disks/default.img
  volatile.initial_source: /var/snap/lxd/common/lxd/disks/default.img
  zfs.pool_name: default  # <---

```